subsys/dm/timeslot_queue.c

```diff
@@ -88,8 +87,7 @@ int timeslot_queue_append(struct dm_request *req, uint32_t start_ref_tick,
 		last = SYS_SLIST_PEEK_TAIL_CONTAINER(&timeslot_list, last, node);
 		list_unlock();
 
-		distance = time_distance_get(last->timeslot_req.start_time, start_time);
-		if (distance < US_TO_RTC_TICKS(last->timeslot_req.timeslot_length_us +
+		if (start_time < US_TO_RTC_TICKS(last->timeslot_req.timeslot_length_us +
```

Comment:
I would interpret the previous check as
difference between last start and new start compared to time in old slot plus minimum slot
which I expect mean if delta from last time fits in old slot, plus there is time to execute slot.

This would be strange indeed as I except this code checks (since return is busy) existence of an overlap.
In this case any start falling between last start and last start plus slot length would be an overlap.

Why then new start is compared to old length plus min length? I would expect this should be old start plus old length (or at least min length).

What is proposed seems odd to me (but I don't know timeslots so it might be that there are some assumptions needed to understand it that I miss)

Reply:
yes you are right, this does not make sense.